### PR TITLE
Custom exception for buffer over/underflow

### PR DESCRIPTION
**Pull request by msx80**
Implements #872

### CODE REVIEW

**Comment by theigl** [comment]
@msx80: Thanks a lot for the PR! 

Please apply `mvn formatter:format` on the PR, so we can be sure it adheres to the project code style.

Since this is affecting the public API, I'll try to get feedback from the project's original author.

File: test/com/esotericsoftware/kryo/io/ByteBufferInputOutputTest.java [hunk 1]
```diff
@@ -87,4 +89,36 @@ void testByteBufferOutputPosition () {
 		assertEquals(10, inputBuffer.readInt());
 		assertEquals(9, byteBuffer.position());
 	}
+	
+		@Test
+	void testOverflow () {
```
Comment (theigl) [inline test/com/esotericsoftware/kryo/io/ByteBufferInputOutputTest.java:94]: Please use JUnit 5 style assertions for exceptions:

https://stackoverflow.com/a/40268447/441266

**Review by theigl — CHANGES_REQUESTED** [review]

**Comment by msx80** [comment]
should be ok now

**Comment by theigl** [comment]
@msx80: Thank you for your patience! 

I finally found the time to get some feedback. We only need one minor change to your PR: Please remove the `KryoIOException` superclass. We don't really need it at the moment and can add it back anytime, should somebody ask for it. Then I'll merge the PR.

**Comment by msx80** [comment]
i've removed it, let me know if it's ok thank you :)

**Review by theigl — APPROVED** [review]

**Comment by theigl** [comment]
Merged, thank you @msx80!

**Pull request merged**